pkgs/by-name/op/opencloud-desktop/package.nix

```
@@ -0,0 +1,68 @@
+{
+  lib,
+  cmake,
+  kdePackages,
+  fetchFromGitHub,
+  libre-graph-api-cpp-qt-client,
+  kdsingleapplication,
+  nix-update-script,
+  versionCheckHook,
+  stdenv,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "opencloud-desktop";
+  version = "1.0.0";
+  src = fetchFromGitHub {
+    owner = "opencloud-eu";
+    repo = "desktop";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-sGbjFPidPncCu9LqaeClrXoKQUzhbR1XbX8RoLuz+N8=";
+  };
+
+  # exporting this envars for cmake to pick up the directories where both ECM and qt6keychain are
+  prePatch = ''
+    export ECM_DIR=${kdePackages.extra-cmake-modules}
+    export Qt6Keychain_DIR=${kdePackages.qtkeychain}
+  '';
+  buildInputs =
+    [
+      kdePackages.qtbase
+      kdePackages.qtdeclarative
+      kdePackages.qt6ct
+      kdePackages.qttools
+      kdePackages.qtgraphs
+      kdePackages.qtnetworkauth
+      kdePackages.qtstyleplugin-kvantum
+      libre-graph-api-cpp-qt-client
+      kdsingleapplication
+    ]
+    ++ lib.optionals (stdenv.hostPlatform.isLinux) [
+      kdePackages.qtwayland
+    ];
+
+  nativeBuildInputs = [
+    cmake
+    kdePackages.qtkeychain
+    kdePackages.qtwebengine
+    kdePackages.extra-cmake-modules
+    kdePackages.wrapQtAppsHook
+  ];
+
+  strictDeps = true;
+  wrapGApps = false;
+  doInstallCheck = true;
+  nativeInstallCheckInputs = [ versionCheckHook ];
+  versionCheckProgram = "${placeholder "out"}/bin/opencloudcmd";
+  versionCheckProgramArg = "--version";
```

nit: spacing these out would make them more readable
```suggestion
  strictDeps = true;
  
  wrapGApps = false;
  
  doInstallCheck = true;
  nativeInstallCheckInputs = [ versionCheckHook ];
  versionCheckProgram = "${placeholder "out"}/bin/opencloudcmd";
  versionCheckProgramArg = "--version";
```

---

yep makes sense :)